src/Uno.UWP/ApplicationModel/Background/SystemTrigger.Android.cs

```diff
@@ -0,0 +1,35 @@
+#if __ANDROID__
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Windows.ApplicationModel.Background
+{
+	public partial class SystemTrigger : IBackgroundTrigger
+	{
+		public  bool OneShot { get; }
+		public SystemTriggerType TriggerType { get; }
+		public SystemTrigger( SystemTriggerType triggerType,  bool oneShot) 
+		{
+			OneShot = oneShot;
+			TriggerType = triggerType;
+			switch(triggerType)
```

Comment:
I think this kind of validation would be best to perform at the receiver site, not inside this class. Is it what UWP does ?

Reply:
As SystemTriggerType enum has all possible values, and only some of them are implemented in Android (PR#2655), somewhere this have to be checked.
I choose this place - as soon in code as it is possible.
If this check would be in BackgroundTaskBuilder.SetTrigger, then this PR could be not Android specific, but for all platforms.
Should I move this check to BackgroundTaskBuilder and make this PR all platform?

Reply:
If UWP does not check on this object's creation, yes it would make more sense to move in the builder, yes.

Reply:
UWP supports all types, so UWP doesn't check anywhere :)
Ok, I will move checking to  BackgroundTaskBuilder, and make this PR all platform.